src/gasPriceOracle/adapters/ethereum-viem.ts

```diff
@@ -1,8 +1,16 @@
 import { PublicClient } from "viem";
 import { InternalGasPriceEstimate } from "../types";
+import { BigNumber } from "../../utils";
 
-export function eip1559(provider: PublicClient, _chainId: number): Promise<InternalGasPriceEstimate> {
-  return provider.estimateFeesPerGas();
+export async function eip1559(
+  provider: PublicClient,
+  _chainId: number,
+  baseFeeMultiplier: number
+): Promise<InternalGasPriceEstimate> {
+  const { maxFeePerGas: _maxFeePerGas, maxPriorityFeePerGas } = await provider.estimateFeesPerGas();
+  const maxFeePerGasScaled = BigNumber.from(_maxFeePerGas.toString()).mul(baseFeeMultiplier);
+  const maxFeePerGas = BigInt(maxFeePerGasScaled.toString()) + maxPriorityFeePerGas;
+  return { maxFeePerGas, maxPriorityFeePerGas };
```

Comment:
These should be BigInt types, not BigNumber. `getViemGasPriceEstimate()` handles BigInt -> BigNumber conversions centrally. That will make it easier for us to eventually transition BigNumber -> BigInt.

One additional thought: since these adapters return both `maxFeePerGas` and `maxPriorityFeePerGas`, we can actually just apply the base fee padding centrally in `getViemGasPriceEstimate()` anyway. That should mean we don't need to touch any of these adapters.

Example:

```
({ maxFeePerGas, maxPriorityFeePerGas } = await gasPriceFeeds[chainId](...));
maxFeePerGas = (maxFeePerGas - maxPriorityFeePerGas) * baseMultiplier + maxPriorityFeePerGas;
```

Reply:
I don't love applying the multiplier in the getViemPriceEstimate function because it means backing out the priority fee. And some chains might not always add the priority fee to the base fee. This also would diverge from how the ethers adapters work

Reply:
Yeah, per subsequent comment it would ideally have been done in`getGasEstimate()`. Problem is that it won't work properly for Linea :\

Reply:
Hm why wouldn't it work for linea? In the linea adapter we'd return the priority fee and the base fee and then in the top level function getGasPriceEstimate we could back out the priority fee?